driver/parallel.go

```diff
@@ -0,0 +1,222 @@
+package driver
+
+import (
+	"sync"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/expr"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/scanner"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng"
+)
+
+type parallelHead struct {
+	proc.Base
+	once sync.Once
+	pg   *parallelGroup
+
+	mu sync.Mutex // protects below
+	sc ScannerCloser
+}
+
+func (ph *parallelHead) closeOnDone() {
+	select {
+	case <-ph.Context.Done():
+		ph.mu.Lock()
+		defer ph.mu.Unlock()
+		if ph.sc != nil {
+			ph.sc.Close()
+			ph.sc = nil
+		}
+	}
+}
+
+func (ph *parallelHead) Pull() (zbuf.Batch, error) {
+	// Trigger the parallel group to read from the multisource.
+	ph.pg.once.Do(func() {
+		go ph.pg.run()
+	})
+	// Ensure open scanners are closed when flowgraph execution stops.
+	ph.once.Do(func() {
+		go ph.closeOnDone()
+	})
```

Comment:
Nit: No need for parallelGroup.once.
```suggestion
	ph.once.Do(func() {
		go ph.closeOnDone()
		go ph.pg.run()
	})
```

Reply:
pg.once exists because pg.run needs to run whenever the first (of possibly many) parallel heads get their Pull.